src/nimblepkg/packageparser.nim

```diff
@@ -258,10 +258,16 @@ proc readPackageInfoFromNimble(path: string; result: var PackageInfo) =
             for i in ev.value.multiSplit:
               if i.splitFile().ext == ".nim":
                 raise newException(NimbleError, "`bin` entry should not be a source file: " & i)
+              let (binSrc, bin) = if '=' notin i: (i, i) else:
+                let spl = i.split('=', 1)
+                (spl[0], spl[1])
               if result.backend == "js":
-                result.bin.add(i.addFileExt(".js"))
+                result.bin.add(bin.addFileExt(".js"))
               else:
-                result.bin.add(i.addFileExt(ExeExt))
+                result.bin.add(bin.addFileExt(ExeExt))
+              if dirExists(result.binDir / result.bin[^1]):
+                result.bin[^1] &= ".out"
```

Comment:
huh? Why this weird code?

```suggestion
                result.bin.add(".out")
```

Reply:
Basically issue #308, if binary file has same name as a directory, it won't get installed so we add a `.out` to the binary name.

Reply:
This also got cleaned up in the update. I still want to improve the `.out` handling cause `nimble run a` doesn't work if it is compiled as `a.out`.